store/CSDK/test/main_ft.cpp

```diff
@@ -0,0 +1,1172 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <jni.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include "../src/CarbonReader.h"
+#include "../src/CarbonRow.h"
+#include "../src/CarbonWriter.h"
+#include "../src/CarbonSchemaReader.h"
+#include "../src/Schema.h"
+#include "../src/CarbonProperties.h"
+#include "gtest/gtest.h"
+
+using namespace std;
+
+JavaVM *jvm;
+JNIEnv *env;
+int my_argc;
+char** my_argv;
+
+/**
+ * init jvm
+ *
+ * @return
+ */
+JNIEnv *initJVM() {
+    JNIEnv *env;
+    JavaVMInitArgs vm_args;
+    int parNum = 2;
+    int res;
+    JavaVMOption options[parNum];
+
+    options[0].optionString = "-Djava.class.path=../../sdk/target/carbondata-sdk.jar";
+    options[1].optionString = "-verbose:jni";                // For debug and check the jni information
+    //    options[2].optionString = "-Xmx12000m";            // change the jvm max memory size
+    //    options[3].optionString = "-Xms5000m";             // change the jvm min memory size
+    //    options[4].optionString = "-Djava.compiler=NONE";  // forbidden JIT
+    vm_args.version = JNI_VERSION_1_8;
+    vm_args.nOptions = parNum;
+    vm_args.options = options;
+    vm_args.ignoreUnrecognized = JNI_FALSE;
+
+    res = JNI_CreateJavaVM(&jvm, (void **) &env, &vm_args);
+    if (res < 0) {
+        fprintf(stderr, "\nCan't create Java VM\n");
+        exit(1);
+    }
+
+    return env;
+}
+
+
+/**
+ * convert Exception to String(Exception toString())
+ * @param jthrowable exc
+ * @return exception as String
+ */
+ char* convertExceptionToString(jthrowable exc){
+        jboolean isCopy = false;
+        jmethodID toString = env->GetMethodID(env->FindClass("java/lang/Object"), "toString", "()Ljava/lang/String;");
+        jstring s = (jstring) env->CallObjectMethod(exc, toString);
+        char *exception_string = (char *) env->GetStringUTFChars(s, JNI_FALSE);
+        return exception_string;
+
+    }
+
+/**
+ * print array result
+ *
+ * @param env JNIEnv
+ * @param arr array
+ */
+void printArray(JNIEnv *env, jobjectArray arr) {
```

Comment:
Can you avoid add the duplicate code with main.cpp?  like: using #include "main.cpp" and invoke the method in this file